plenum/test/common/version/test_version.py

```diff
@@ -0,0 +1,238 @@
+import pytest
+
+from plenum.common.version import (
+    InvalidVersionError, VersionBase, PEP440BasedVersion,
+    SemVerBase, DigitDotVersion, SemVerReleaseVersion,
+    PackageVersion
+)
+
+
+def abc_required(properties=None, classmethods=None):
+    def _f(*_, **__):
+        pass
+
+    res = {}
+    for p in properties or []:
+        res[p] = property(_f)
+    for clm in classmethods or []:
+        res[clm] = classmethod(_f)
+    return res
+
+
+def iterate_abstracts(required, cls_name, base_cls, base_required=None):
+    for i in range(len(required)):
+        _dict = {
+            attr: f for idx, (attr, f) in
+            enumerate(required.items()) if idx <= i
+        }
+        if base_required:
+            _dict.update(base_required)
+        version_cls = type(cls_name, (base_cls,), _dict)
+        yield (version_cls, i + 1 == len(required))
+
+
+@pytest.fixture
+def version_base_required():
+    return abc_required(
+        properties=('full', 'parts', 'release', 'release_parts'),
+        classmethods=('cmp',)
+    )
+
+
+def test_version_base_abstracts(version_base_required):
```

Comment:
What do we test by this test? Why don't we instantiate real instances of Version classes instead of dealing with Abstracts and class internals?